src/solc_0.8/asset/ERC1155ERC721.sol

```diff
@@ -1130,4 +1081,47 @@ contract ERC1155ERC721 is WithSuperOperators, IERC1155, IERC721 {
         _mint(_metadataHash[id & URI_ID], 1, 0, operator, to, newId, "", true);
         emit Extraction(id, newId);
     }
+
+    /// @dev Allows the use of a bitfield to track the initialized status of the version `v` passed in as an arg.
+    /// If the bit at the index corresponding to the given version is already set, revert.
+    /// Otherwise, set the bit and return.
+    /// @param v The version of this contract.
+    function _checkInit(uint256 v) internal {
```

Comment:
Not sure if this is needed, when we make an upgrade, we can ensure we modifiy the init function as we wish.

Reply:
@wighawag I will review this.
My thinking was as follows:
Each time we upgrade, we can modify the init function to suit our needs. 
However, the storage variable `_init` will be set to `true` after the first deployment (which was V1).
Subsequent upgrades would fail if the only requirement was checking that `_init` is `false`.
We could introduce a new storage variable for each upgrade(ie: `_init2`, `_init3`...) , but at the cost of a new storage slot each time.
The other approach I looked at (implemented here) is to use a single uint256 as a bitfield, setting//checking individual bits. This allows us to effectively store 255 bools in one slot.